arch/arm/mach-imx/imx8m/soc.c

```diff
@@ -814,6 +815,13 @@ int boot_mode_getprisec(void)
 {
 	return !!imx8m_detect_secondary_image_boot();
 }
+#else
+int boot_mode_getprisec(void)
+{
+	struct src *psrc = (struct src *)SRC_BASE_ADDR;
+
+	return !!(readl(&psrc->gpr10) & SRC_GPR10_PERSIST_SECONDARY_BOOT);
+}
 #endif
```

Comment:
```suggestion
#else
int boot_mode_getprisec(void)
{
	struct src *psrc = (struct src *)SRC_BASE_ADDR;

	return !!(readl(&psrc->gpr10) & SRC_GPR10_PERSIST_SECONDARY_BOOT);
}
#endif
```
@igoropaniuk I was a bit confused on this one. `boot_mode_getprisec` was included upstream, can we use that instead of what you included here?

Reply:
Technically it does the same, so I think we can just use upstream version

Reply:
I think we can keep the if/else logic because of 1144f3829defd425eb9835bf5d408e018abe7e9f